src/Moat/Types.hs

```diff
@@ -407,7 +388,6 @@ data Options = Options
 --   { typeConstructorModifier = id
 --   , fieldLabelModifier = id
 --   , constructorModifier = id
---   , optionalExpand = False
```

Comment:
We don't use this anywhere and I don't understand its' purpose

Reply:
We don't use it anywhere because in practise, at Mercury at least, people don't want it. Its purpose (the code used to be in shwifty but was removed) is to expand Optionals syntactically. EG by default we write `Maybe String` as `String?` in swift, but we could also write `Optional<String>`. optionalExpand is meant to 'expand' the optional into the more verbose syntax, in case someone wants that. I think it makes sense to provide this functionality, I just never got around to it.

Reply:
Okay, I am on board to implement it. I don't think it would be terribly difficult

Reply:
Yeah, should just be a single `case` in the pretty-printing.